reframe/frontend/autodetect.py

```diff
@@ -151,24 +151,32 @@ def _emit_script_for_pip(job, env):
         ]
         job.prepare(commands, env, trap_errors=True, login=use_login_shell)
 
+    def _emit_custom_script(job, env, commands):
+        job.prepare(commands, env, trap_errors=True, login=use_login_shell)
+
     getlogger().info(
         f'Detecting topology of remote partition {part.fullname!r}: '
         f'this may take some time...'
     )
     topo_info = {}
     try:
         prefix = runtime.runtime().get_option('general/0/remote_workdir')
+        custom_command = runtime.runtime().get_option(
+            'general/0/remote_command'
+        )
         with _copy_reframe(prefix) as (dirname, use_pip):
             with osext.change_dir(dirname):
                 job = Job.create(part.scheduler,
                                  part.launcher_type(),
                                  name='rfm-detect-job',
                                  sched_access=part.access)
-                if use_pip:
-                    _emit_script_for_pip(job, [part.local_env])
+                if custom_command:
+                    _emit_custom_script(job, [part.local_env], custom_command)
                 else:
-                    _emit_script_for_source(job, [part.local_env])
-
+                    if use_pip:
+                        _emit_script_for_pip(job, [part.local_env])
+                    else:
+                        _emit_script_for_source(job, [part.local_env])
```

Comment:
```suggestion
                elif use_pip:
                    _emit_script_for_pip(job, [part.local_env])
                else:
                    _emit_script_for_source(job, [part.local_env])
```
No need to indent this further I think